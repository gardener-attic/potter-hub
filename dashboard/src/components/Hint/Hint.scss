.extraClass {
  pointer-events: auto;
  text-align: center;
  background: #1c2b39;
  &:hover {
    visibility: visible;
    opacity: 1;
  }
  &:after {
    border-top-color: #1c2b39;
    border-top-style: solid;
    border-top-width: 6px;
  }
}
