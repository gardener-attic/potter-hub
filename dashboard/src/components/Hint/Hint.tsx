import * as React from "react";
import { HelpCircle } from "react-feather";
import ReactTooltip from "react-tooltip";

import "./Hint.css";

interface IHintProps {
  reactTooltipOpts?: any;
  id?: string;
}

export class Hint extends React.Component<IHintProps> {
  public render() {
    // Generate a random ID if not given
    const id =
      this.props.id ||
      Math.random()
        .toString(36)
        .substring(7);
    return (
      <React.Fragment>
        <a data-tip={true} data-for={id}>
          <HelpCircle className="icon" color="white" fill="#5F6369" />
        </a>
        <ReactTooltip
          id={id}
          className="extraClass"
          effect="solid"
          {...this.props.reactTooltipOpts}
        >
          {this.props.children}
        </ReactTooltip>
      </React.Fragment>
    );
  }
}

export default Hint;
