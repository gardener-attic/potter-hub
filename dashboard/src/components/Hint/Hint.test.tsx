import { shallow } from "enzyme";
import * as React from "react";
import ReactTooltip from "react-tooltip";
import Hint from ".";

it("should render a Hint with additional props and children", () => {
  const wrapper = shallow(
    <Hint reactTooltipOpts={{ foo: "bar" }} id="foobar">
      <p>this is a test!</p>
    </Hint>,
  );
  const tooltip = wrapper.find(ReactTooltip);
  expect(tooltip).toExist();
  expect(tooltip.props()).toMatchObject({ foo: "bar" });
  expect(wrapper).toMatchSnapshot();
});
