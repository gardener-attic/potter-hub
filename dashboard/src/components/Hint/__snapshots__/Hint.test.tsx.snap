// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Hint with additional props and children 1`] = `
<Fragment>
  <a
    data-for="foobar"
    data-tip={true}
  >
    <HelpCircle
      className="icon"
      color="white"
      fill="#5F6369"
      size="24"
    />
  </a>
  <ReactTooltip
    className="extraClass"
    clickable={false}
    effect="solid"
    foo="bar"
    id="foobar"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  >
    <p>
      this is a test!
    </p>
  </ReactTooltip>
</Fragment>
`;
